src/System.Text.Primitives/System/Text/Formatting/Internal/IntegerFormatter.Utf16.cs

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Text
+{
+    internal static class InvariantUtf16IntegerFormatter
+    {
+        private const char Minus = '-';
+        private const char Period = '.';
+        private const char Seperator = ',';
+
+        // Invariant formatting uses groups of 3 for each number group seperated by commas.
+        //   ex. 1,234,567,890
+        private const int GroupSize = 3;
+
+        public static bool TryFormatDecimalInt64(long value, byte precision, Span<byte> buffer, out int bytesWritten)
+        {
+            int digitCount = FormattingHelpers.CountDigits(value);
+            int charsNeeded = digitCount + (int)((value >> 63) & 1);
+            Span<char> span = buffer.NonPortableCast<byte, char>();
```

Comment:
>span [](start = 23, length = 4)

Is the non-portable cast needed? Could you instead divide the length by 2 (plus maybe check for odd number of bytes), and then just pin the buffer and cast the pointer to char*